File: src/main/java/bbarena/model/event/roster/RosterEvent.java
package bbarena.model.event.roster;

import bbarena.model.Arena;
import bbarena.model.event.Event;
import bbarena.model.team.Player;

public abstract class RosterEvent extends Event {

	private static final long serialVersionUID = 8447866046452336969L;
	protected long _playerId = 0;

	public RosterEvent(long playerId) {
		super();
		_playerId = playerId;
	}

	public long getPlayerId() {
		return _playerId;
	}

	public Player getPlayer(Arena arena) {
		return arena.getPlayerManager().getPlayer(_playerId);
	}

}

File: src/main/kotlin/bbarena/model/event/roster/RosterEvent.kt
package bbarena.model.event.roster

import bbarena.model.Arena
import bbarena.model.event.Event
import bbarena.model.team.Player

abstract class RosterEvent(playerId: Long) : Event() {
    var playerId: Long = 0
        protected set

    init {
        this.playerId = playerId
    }

    fun getPlayer(arena: Arena): Player? {
        return arena.playerManager.getPlayer(playerId)
    }

    companion object {

        private val serialVersionUID = 8447866046452336969L
    }

}